Lib/fontTools/varLib/instancer/__init__.py

```diff
@@ -250,6 +252,50 @@ def __post_init__(self):
                 f"minimum={self.minimum:g}, default={self.default:g}, maximum={self.maximum:g})"
             )
 
+    def reverse_negate(self):
+        v = self
+        return self.__class__(-v[2], -v[1], -v[0], v[4], v[3])
+
+    def normalizeValue(self, v, extrapolate=True):
+        lower, default, upper, distanceNegative, distancePositive = self
+        assert lower <= default <= upper
+
+        if not extrapolate:
+            v = max(lower, min(upper, v))
+
+        if v == default:
+            return 0
+
+        if default < 0:
+            return -self.reverse_negate().normalizeValue(-v, extrapolate=extrapolate)
+
+        # default >= 0 and v != default
+
+        if v > default:
+            return (v - default) / (upper - default)
+
+        # v < default
+
+        if lower >= 0:
+            return (v - default) / (default - lower)
+
+        # lower < 0 and v < default
```

Comment:
I see that your example make sense but I'm still not 100% clear why this situation in which default >=0 and v < default and lower < 0 needs to take into account the positive/negative distances, whereas the other cases do not. Is it because in this particular case the value `v` falls in a segment that may span both the negative and positive sides and thus the ratio between these two matters?

Reply:
> Is it because in this particular case the value v falls in a segment that may span both the negative and positive sides and thus the ratio between these two matters?

Correct.